src/geo_index.c

```diff
@@ -5,28 +5,30 @@
 
 /* Add a docId to a geoindex key. Right now we just use redis' own GEOADD */
 int GeoIndex_AddStrings(GeoIndex *gi, t_docId docId, const char *slon, const char *slat) {
+  int rv = REDISMODULE_OK;
 
   RedisModuleString *ks = IndexSpec_GetFormattedKey(gi->ctx->spec, gi->sp, INDEXFLD_T_GEO);
   RedisModuleCtx *ctx = gi->ctx->redisCtx;
 
   /* GEOADD key longitude latitude member*/
   RedisModuleCallReply *rep = RedisModule_Call(ctx, "GEOADD", "sccl", ks, slon, slat, docId);
-  if (rep == NULL) {
-    return REDISMODULE_ERR;
-  }
+
   int repType = RedisModule_CallReplyType(rep);
-  RedisModule_FreeCallReply(rep);
-  if (repType == REDISMODULE_REPLY_ERROR) {
-    return REDISMODULE_ERR;
-  } else {
-    return REDISMODULE_OK;
+  if (rep == NULL || repType == REDISMODULE_REPLY_ERROR) {
+    rv = REDISMODULE_ERR;
   }
+
+  RedisModule_FreeCallReply(rep);
+  return rv;
 }
 
 void GeoIndex_RemoveEntries(GeoIndex *gi, IndexSpec *sp, t_docId docId) {
   RedisModuleString *ks = IndexSpec_GetFormattedKey(sp, gi->sp, INDEXFLD_T_GEO);
   RedisModuleCtx *ctx = gi->ctx->redisCtx;
   RedisModuleCallReply *rep = RedisModule_Call(ctx, "ZREM", "sl", ks, docId);
+  if (rep == NULL) {
```

Comment:
Reply might also be REDISMODULE_ERR but I do not think its that important

Reply:
What if we give it a wrong number of args or internal error occur? its better to be on the safe side.